test/ringpop-assert.js

```diff
@@ -625,6 +625,59 @@ function assertStats(t, tc, statusCountsOralive, suspect, faulty, members) {
     return result;
 }
 
+function assertStateChange(t, tc, addressOrIndex, status, expectedDuration, allowedJitter) {
```

Comment:
This function measures the time from the first join but is used in both tests (state in join request and state updated after the join completed). Might be better to make that configurable I think and configure it correctly in both tests. This might reduce flaky-ness of the jitter.